### PR TITLE
Topic/doxygentomd astro docs

### PR DESCRIPTION
**Pull request by gnu-lorien**
I've found fewer errors in the generated pages doing it this way, but it's not 1-to-1. The two biggest changes here are:
- All of the extra stuff in all because moxygen just didn't generate things for files and directories
- The RHAPI_ pages are links to anything like a struct or a class that is fully documented somewhere else. That means some of these pages are just lists of links, and others have some enums and such.

### CODE REVIEW

**Comment by nonathaj** [comment]
Up until now, we've avoided generating docs for the RHAPI stuff in the RallyHereAPI or RallyHereDeveloperAPI plugins.  Partly for the reasons you listed - that many of the contents are just links.  The API plugins are autogenerated - which means generating documentation for it will end up being largely identical to the generated docs for every API endpoint.

Instead we just were trying to generate reference docs for the RallyHereIntegration plugin - which contains all of our manually created code.

File: Documentation/astromd/all/CrossplayInput_8h.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+
+### `ERHAPI_CrossplayInput` <a id="CrossplayInput_8h_1a7bb2cd98a11e06372454b8e43954aee4"></a>
+Crossplay input preferences. Used to determine whether players can queue together, and the pools of players that are used to generate matches.
+
+
+
+| Enumerator | Initializer|
+|------------|------------|
+|`UMETA`|`=(ToolTip="No crossplay input preferences. Can play with players using any input")`|
```
Comment (nonathaj) [inline Documentation/astromd/all/CrossplayInput_8h.md:26]: Looks like this processed the macro after the enum as the actual enum contents.  Not sure if this is a doxygen or converter issue.

Comment (gnu-lorien) [reply Documentation/astromd/all/CrossplayInput_8h.md:26]: Looks like that comes straight from the Doxygen. The syntax those macros create has confused the hell out of it.

```
      <sectiondef kind="enum">
      <memberdef kind="enum" id="CrossplayInput_8h_1a7bb2cd98a11e06372454b8e43954aee4" prot="public" static="no" strong="yes">
        <type>uint8</type>
        <name>ERHAPI_CrossplayInput</name>
        <enumvalue id="CrossplayInput_8h_1a7bb2cd98a11e06372454b8e43954aee4aa3f80d8f0f1b4ffa8c475f5d60ea5635" prot="public">
          <name>UMETA</name>
          <initializer>=(ToolTip=&quot;No crossplay input preferences. Can play with players using any input&quot;)</initializer>
          <briefdescription>
          </briefdescription>
          <detaileddescription>
          </detaileddescription>
        </enumvalue>
        <enumvalue id="CrossplayInput_8h_1a7bb2cd98a11e06372454b8e43954aee4aa3f80d8f0f1b4ffa8c475f5d60ea5635" prot="public">
          <name>UMETA</name>
          <initializer>=(ToolTip=&quot;Can only play with players using a keyboard and mouse&quot;)</initializer>
          <briefdescription>
          </briefdescription>
          <detaileddescription>
          </detaileddescription>
        </enumvalue>
        <enumvalue id="CrossplayInput_8h_1a7bb2cd98a11e06372454b8e43954aee4aa3f80d8f0f1b4ffa8c475f5d60ea5635" prot="public">
          <name>UMETA</name>
          <initializer>=(ToolTip=&quot;Can only player with players using a gamepad/controller&quot;)</initializer>
          <briefdescription>
          </briefdescription>
          <detaileddescription>
          </detaileddescription>
        </enumvalue>
        <enumvalue id="CrossplayInput_8h_1a7bb2cd98a11e06372454b8e43954aee4aa3f80d8f0f1b4ffa8c475f5d60ea5635" prot="public">
          <name>UMETA</name>
          <initializer>=(ToolTip=&quot;Can only player with players using a touch screen&quot;)</initializer>
          <briefdescription>
          </briefdescription>
          <detaileddescription>
          </detaileddescription>
        </enumvalue>
        <briefdescription>
<para>Crossplay input preferences. Used to determine whether players can queue together, and the pools of players that are used to generate matches. </para>
        </briefdescription>
        <detaileddescription>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="/home/asayman/UnrealDev/unreal-plugins/RallyHereAPI/Source/RallyHereAPI/Public/CrossplayInput.h" line="19" column="1" bodyfile="/home/asayman/UnrealDev/unreal-plugins/RallyHereAPI/Source/RallyHereAPI/Public/CrossplayInput.h" bodystart="20" bodyend="25"/>
      </memberdef>
```

**Comment by gnu-lorien** [comment]
> Up until now, we've avoided generating docs for the RHAPI stuff in the RallyHereAPI or RallyHereDeveloperAPI plugins. Partly for the reasons you listed - that many of the contents are just links. The API plugins are autogenerated - which means generating documentation for it will end up being largely identical to the generated docs for every API endpoint.
> 
> Instead we just were trying to generate reference docs for the RallyHereIntegration plugin - which contains all of our manually created code.

That's part of why I wasn't entirely sure that this would be considered an improvement. It wasn't clear to me whether that's intentional, or something you had to do because of moxygen.

I was thinking that if the desire here really is "just the group pages and nothing else" that I could make a mode in the count that outputs everything listed in group pages as if they were the source and none of the other stuff. I don't think that would be *too* hard other than we might generate links to pages that weren't generated at all.

**Comment by nonathaj** [comment]
Leaving this as-is is probably fine for the time being then :+1:

File: Documentation/astromd/all/CrossplayInput_8h.md (hunk 1 above)
Comment (hirez-cranney) [reply Documentation/astromd/all/CrossplayInput_8h.md:26]: Is doxygen able to understand specific macros being noop and just ignoring it?

**Pull request closed without merging**